share/rocm/cmake/ROCMVersionDLL.cmake

```diff
@@ -0,0 +1,58 @@
+# ######################################################################################################################
+# Copyright (C) 2022 Advanced Micro Devices, Inc.
+# ######################################################################################################################
+
+set(_ROCMVersionDLL_LIST_DIR "${CMAKE_CURRENT_LIST_DIR}")
+
+define_property(
+    TARGET PROPERTY DLL_FILE_VERSION
+    BRIEF_DOCS "This target property records the version of the generated DLL file."
+    FULL_DOCS "The value stored in this property will be set as the FileVersion property of the DLL."
+)
+
+set(ROCM_DLL_VERSION "$ENV{ROCM_DLL_VERSION}" CACHE STRING "The version to add to DLLs by default.")
+set(ROCM_DLL_FILE_VERSION "$ENV{ROCM_DLL_FILE_VERSION}" CACHE STRING "The version of the file to attach to DLLs.")
```

Comment:
I don't think this makes sense as a global option. It's really a property of the library itself and I don't think we want to bake in the assumption that there's only one library built per CMake invokation. There's a lot of ways that assumption could be violated, e.g., building libraries via `add_subdirectory`.